jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3.objectActions
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.LangDataKeys
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.ui.Messages
+import com.intellij.ui.AnActionButton
+import org.jetbrains.annotations.TestOnly
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.Delete
+import software.amazon.awssdk.services.s3.model.DeleteObjectsRequest
+import software.amazon.awssdk.services.s3.model.ObjectIdentifier
+import software.aws.toolkits.core.telemetry.TelemetryNamespace
+import software.aws.toolkits.jetbrains.core.AwsClientManager
+import software.aws.toolkits.jetbrains.services.s3.S3VirtualBucket
+import software.aws.toolkits.jetbrains.services.s3.S3VirtualDirectory
+import software.aws.toolkits.jetbrains.services.s3.bucketEditor.S3KeyNode
+import software.aws.toolkits.jetbrains.services.s3.bucketEditor.S3TreeTable
+import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+import javax.swing.tree.DefaultMutableTreeNode
+
+class DeleteObjectAction(
+    private var treeTable: S3TreeTable,
+    val bucket: S3VirtualBucket
+
+) : AnActionButton("Delete Object", null, AllIcons.Actions.Cancel), TelemetryNamespace {
```

Comment:
Remove the extra empty line.

Why do we use `AnActionButton` for all the object related actions? We should use [ActionWrappers](https://github.com/aws/aws-toolkit-jetbrains/blob/master/jetbrains-core/src/software/aws/toolkits/jetbrains/components/telemetry/ActionWrappers.kt) for all our actions ideally. I think it's better to add another wrapper for `AnActionButton` if we have to use it.

Reply:
Editor popup menu wasn't able to register object actions with AnActionWrapper. I think I can try to create a separate wrapper for AnActionButton and have the telemetry reports in there.